static/app/views/settings/components/teamSelect.tsx

```diff
@@ -92,6 +92,32 @@ function TeamSelect({
   const {teams, onSearch, fetching} = useTeams();
   const {orgRoleList, teamRoleList} = organization;
 
+  const slugsToFilter: string[] =
+    selectedTeams?.map(tm => tm.slug) || selectedTeamRoles?.map(tm => tm.teamSlug) || [];
+
+  // determine if adding a team changes the minimum team role
+  const getTopOrgRole = (): string | undefined => {
```

Comment:
this has some different logic because it has to take into account the "staged" changes on the `organizationMemberDetail` page. you can add a team that would change your minimum team role but not save

Reply:
if you add or remove teams and that would cause some change in your minimum team role, we can't rely on the backend to calculate that